src/main/java/no/unit/bibs/contents/DBClient.java

```diff
@@ -155,32 +155,33 @@ public String getContents(String isbn) throws NotFoundException {
      * Updates the contentsDocument identified by its isbn.
      *
      * @param document contentsDocument to update
+     * @return
      * @throws CommunicationException exception while connecting to database
      */
-    protected void updateContents(ContentsDocument document) throws CommunicationException {
+    protected UpdateItemResponse updateContents(ContentsDocument document) throws CommunicationException {
         try {
-            HashMap<String, AttributeValue> keyToUpdate = new HashMap<>();
+            Map<String, AttributeValue> keyToUpdate = new HashMap<>();
             keyToUpdate.put(PRIMARYKEY_ISBN, AttributeValue.builder().s(document.getIsbn()).build());
             Map<String, AttributeValueUpdate> attributeUpdates = this.findValuesToUpdate(document);
             UpdateItemRequest updateItemRequest = UpdateItemRequest
-                    .builder()
-                    .key(keyToUpdate)
-                    .tableName(tableName)
-                    .attributeUpdates(attributeUpdates)
-                    .build();
-            dbClient.updateItem(updateItemRequest);
-            logger.info("contents updated");
+                .builder()
+                .key(keyToUpdate)
+                .tableName(tableName)
+                .returnValues("ALL_NEW")
+                .attributeUpdates(attributeUpdates)
+                .build();
+            return dynamoDbClient.updateItem(updateItemRequest);
+
         } catch (Exception e) {
-            logger.error(e.getMessage(), e);
             throw new CommunicationException("Update error: " + e.getMessage(), e);
```

Comment:
Here you keep CommunicationException but another place you remove it?

Reply:
reverted back to original